src/git.rs

```diff
@@ -78,65 +81,144 @@ impl Git {
+        }
 
-            if let Some(v) = author.name() {
-                self.update_str(COMMIT_AUTHOR, v.to_string());
-            }
-            let status_file = Self::git2_dirty_stage(&repo);
-            if status_file.trim().is_empty() {
-                self.update_bool(GIT_CLEAN, true);
-            } else {
-                self.update_bool(GIT_CLEAN, false);
-            }
-            self.update_str(GIT_STATUS_FILE, status_file);
+        if let Some(v) = author.name() {
+            self.update_str(COMMIT_AUTHOR, v.to_string());
         }
+        let status_file = Self::git2_dirty_stage(&repo);
+        if status_file.trim().is_empty() {
+            self.update_bool(GIT_CLEAN, true);
+        } else {
+            self.update_bool(GIT_CLEAN, false);
+        }
+        self.update_str(GIT_STATUS_FILE, status_file);
+
+        Ok(())
+    }
+
+    #[cfg(feature = "gitoxide")]
+    fn init_gitoxide(&mut self, path: &Path) -> SdResult<()> {
+        let repo = gix::discover(path).map_err(ShadowError::new)?;
+
+        let branch = repo
+            .head_name()
+            .map_err(ShadowError::new)?
+            .map(|n| n.shorten().to_string())
+            .unwrap_or_default();
+
+        // TODO: how does the --contains option actually work?
+        let refs = repo.references().map_err(ShadowError::new)?;
+        let tag = refs
+            .tags()
+            .map_err(ShadowError::new)?
+            .filter_map(|tag| match tag {
+                Ok(mut tag) => {
+                    let peeled_tag_id = match tag.inner.peeled {
+                        Some(id) => id,
+                        None => {
+                            tag.peel_to_id_in_place().ok()?;
+                            tag.inner.peeled.expect("peeled")
+                        }
+                    };
+                    repo.head_id().ok()?.eq(&peeled_tag_id).then_some(tag)
+                }
+                _ => None,
+            })
+            .map(|tag| tag.name().shorten().to_string())
+            .collect::<Vec<_>>()
+            .join("\n");
+
+        let last_tag = repo
+            .head_commit()
+            .map_err(ShadowError::new)?
+            .describe()
+            .names(gix::commit::describe::SelectRef::AllTags)
+            .format()
+            .map_err(ShadowError::new)?;
+
+        // TODO: add .hex_len(0) option (git: --abbrev=0) in `Format`s `Display` implementation
+        let last_tag = if let Some(name) = last_tag.name.as_deref() {
+            name.to_string()
+        } else {
+            String::new()
+        };
+
+        self.update_str(BRANCH, branch);
+        self.update_str(TAG, tag);
+        self.update_str(LAST_TAG, last_tag);
+
+        let commit = repo.head_commit().map_err(ShadowError::new)?;
+
+        self.update_str(COMMIT_HASH, commit.id().to_string());
+        self.update_str(
+            SHORT_COMMIT,
+            commit
+                .id()
+                .shorten()
```

Comment:
Isn't there a `shorten_or_id()` method for exactly that? The `git2` implementations seems to always use the full ID at least whereas this one defaults to empty strings if shortening fails.